libraries/AP_Scripting/lua_bindings.cpp

```diff
@@ -250,11 +250,196 @@ const luaL_Reg AP_Logger_functions[] = {
     {NULL, NULL}
 };
 
+
+static int add_param(lua_State *L) {
+    check_arguments(L, 3, "param.add");
+
+    if (lua_type(L, 3) != LUA_TTABLE) {
+        return luaL_error(L, "expected param table for argument 3");
+    }
+
+    size_t num_params = lua_rawlen(L,3);
+    if (num_params == 0) {
+        return luaL_error(L, "need at least one param");
+    }
+
+    uint16_t index = static_cast<uint16_t>(luaL_checkinteger(L, 1));
+
+    // this magic value is to be compared against a automaticaly loaded param at the start of the new table,
+    // the the magics don't match we should reset the new table to defaults 
+    //const uint16_t magic = static_cast<uint16_t>(luaL_checkinteger(L, 2));
+
+    AP_Param::Info *info = (AP_Param::Info *)hal.util->malloc_type(sizeof(AP_Param::Info)*(num_params+1), AP_HAL::Util::Memory_Type::MEM_FAST);
```

Comment:
fairly sure this malloc is not the right thing to do? Do I also need to free rather than just delete?

Reply:
You use free if you are just malloc'ing, delete is when you are removing something you newed. Also is this supposed to be system memory vs lua sandboxed memory?

Reply:
(you can also use the placement with new to dictate where it goes if you want to use the sandboxed memory)